sui_core/src/authority.rs

```diff
@@ -546,6 +560,37 @@ impl AuthorityState {
             }
         }
 
+        // If a checkpoint store is present, ensure it is up-to-date with the latest
+        // batches.
+        if let Some(checkpoint) = &state._checkpoints {
+            let next_expected_tx = checkpoint.next_transaction_sequence_expected();
+
+            // Get all unprocessed checkpoints
```

Comment:
This may happen if there is a crash between committing a batch, and send it it for processing to a checkpoint. Since batch writting and checkpoint writting are not atomic it is possible that the first happens and the second does not happen.